### DISCUSSION TITLE
Check role before delete

### DISCUSSION THREAD

**Discussion by Logical-Developer** [spatie/laravel-permission#1821] [Q&A] [answered]
Hi
How can I check role doesn't have any permission before delete ?

**Comment by drbyte** [comment, marked as answer]
I just [added this to the docs](https://spatie.be/docs/laravel-permission/v4/basic-usage/role-permissions) for clarity:

```php
// count the number of permissions assigned to a role
count($role->permissions);
// or
$role->permissions->count();
```
